### ISSUE TITLE
Webpack instead of ruby pl0x

### ISSUE THREAD

**Issue by maxcr** [bigcartel/dugway#158]
I know this is a feature request more than a bug nearly all the libraries in dugway are deprecated. Instead of trying to fix the weird errors I've been getting I've been supplementing dugway with webpack which has actually been working pretty well. But it would be nice if dugway reflected the times and was a bit more dev friendly.

**Comment by johnelliott** [comment]
I didn't have any problems installing/running Dugway.

For other reasons I'm trying my hand at using Webpack also. The most interesting thing on that front that I have found is https://github.com/Shopify/slate/tree/master/packages/slate-liquid-asset-loader.

**Comment by maxcr** [comment]
Slate is garbage. Why I switched to big cartel. Square Space have much better local Dev tools, but BigCartels are still by far the best

**Comment by nabrown** [comment]
@maxcr Ok, off topic now but another site I'm working on is Squarespace-based (not my choice), and since you seem familiar with their dev tools -- when you build your own templates, do you only have control over markup and styles? Or do you actually have programmatic control over data (if/then, loops etc)? And once you depart from an official Squarespace theme, are upgrades manual? 
Any decent resources/docs on this topic?
Thanks in advance if you have time to answer.

**Comment by maxcr** [comment]
@nabrown square space gives the best control over markup and stiles via css/liquid. There are certain divs/classes/I'd which simply cannot be removed without hacky methods like !important. I probably should have been more clear in saying that Shopify's tools, while buggier, and often dubiously implemented, offer more creative freedom. But not the freedom to the degree that is allowed by BigCartel.

**Comment by nabrown** [comment]
Interesting -- thanks for the info. Sounds like nobody has come up with the best developer experience when it comes to these "DIY" e-commmerce platforms. I'm already totally frustrated by the limitations of Squarespace (not having tried developing a full theme yet).

**Comment by maxcr** [comment]
I've been wanting to create a backed that let's people use handlebars. But creating runtimes that are memory safe while handing client credit card numbers isn't easy. Not just that there's so many ways to sploit webapps now... Yeah it's a real pain in the ass

**Comment by johnelliott** [comment]
I applaud both bigcartel and shopify for even offering tools like Dugway and Slate. I'm grateful to have either. A number of smart people evidently put their best effort into building the tools. 👍 

We stopped using bigcartel because 'deploying' by copy/paste into the web editor was to much of a pain: other logged-in users could clobber changes and we had no way to tell what was deployed other than logging out, logging in, downloading and diffing files 🙃

**Comment by maxcr** [comment]
Yeah I switched to WordPress. I had like 2K lines of JS of features that
are standard as part of WordPress. I'll eventually move that JS into a
backend that I can incorporate into ghost blog for an eventually fully
javascript e-commerce code base with all the bells and whistles. 7ish years
since I've used it and I found an easter-egg in the standard theme that
changes all the fonts to Comic Sans. Its almost like they want you to nuke
the default theme and make something from scratch. Product search JS logic
is tied to the mobile navigation? WTF
